In [1]:
!wget http://www.gutenberg.org/files/100/100-0.txt
filePath = "/content/100-0.txt"
ShakespeareFile = open(filePath, "r")
ShakespeareContent = ShakespeareFile.read()
ShakespeareContent = ShakespeareContent.splitlines() # The splitlines() method splits a string into a list. The splitting is done at line breaks.
print(ShakespeareContent[0:50])


--2023-09-29 12:04:07--  http://www.gutenberg.org/files/100/100-0.txt
Resolving www.gutenberg.org (www.gutenberg.org)... 152.19.134.47, 2610:28:3090:3000:0:bad:cafe:47
Connecting to www.gutenberg.org (www.gutenberg.org)|152.19.134.47|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://www.gutenberg.org/files/100/100-0.txt [following]
--2023-09-29 12:04:07--  https://www.gutenberg.org/files/100/100-0.txt
Connecting to www.gutenberg.org (www.gutenberg.org)|152.19.134.47|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5647299 (5.4M) [text/plain]
Saving to: ‘100-0.txt’

100-0.txt           100%[===================>]   5.38M  22.6MB/s    in 0.2s    

2023-09-29 12:04:08 (22.6 MB/s) - ‘100-0.txt’ saved [5647299/5647299]

['\ufeffThe Project Gutenberg eBook of The Complete Works of William Shakespeare, by William Shakespeare', '', 'This eBook is for the use of anyone anywhere in the United States and', 'most other parts of the world 

In [56]:
from tensorflow.keras.preprocessing.text import Tokenizer


t  = Tokenizer()
# Defining 4 document lists
fit_text = ShakespeareContent

t.fit_on_texts(fit_text)
print("total number of tokens", sum(t.word_counts.values()))
print("total number of types", len(t.word_counts))

print(f"most frequent Type in corpus is '{max(t.word_counts)}' and its frequency is '{max(t.word_counts.values())}'")




total number of tokens 972136
total number of types 30491
most frequent Type in corpus is 'the' and its frequency is '30449'


In [64]:
from nltk.stem import PorterStemmer
import nltk
nltk.download('wordnet')
from nltk.stem.wordnet import WordNetLemmatizer


ps = PorterStemmer()
lemmatizer = WordNetLemmatizer()


[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!


In [94]:
words = list(t.word_counts.keys())
stemmedWords = set()
for word in words :
  stemmedWord = ps.stem(word)
  stemmedWords.add(stemmedWord)

print(stemmedWords)
print("total number of types after stemming is ", len(stemmedWords))


{'april', 'phlegmat', 'slack', 'artilleri', 'esteem’d', 'point—', 'curs', 'odiou', 'socrates’', 'have', 'persecutor', 'blink', 'livia', 'pain', 'gravi', 'unqual', 'hammer', 'valentio', 'sprat', '“husband”', '‘hector', '—speak', 'charnel', 'royalti', 'pursuest', 'seaman', 'tyrant', 'bondman’', 'wond’r', 'lodestar', 'waggl', 'rutland', 'slipp’d', '‘great', 'quieter', 'th’affect', "'baa", 'tercel', 'dishclout', 'compass', 'befor', 'unclasp', 'renegado', 'an—fool’', 'chas’d', 'bustl', 'nation—', 'elvish', 'metheglin', 'spoon', 'catastroph', 'debonair', 'sabl', 'cosmo', 'arc', 'quince’', 'whore', 'desirest', 'grammar', 'see—', 'fub', 'industri', 'egypt', 'mankind', 'curtail', 'prank’d', 'haec', 'near—', 'copi', 'demeanour', 'flamen', 'importment’', 'bedew', 'tup', 'mistresship', 'harness’d', 'unvalu', 'o’', 'prithe', 'hers—to', 'furor', 'shortli', 'pui', 'dungeon', 'wayward', 'remerci', 'sway', 'misthought', 'squel', 'cheapsid', 'suck', 'harsher', 'corn', 'wilder', 'ever—a', 's’', 'page', '

In [97]:


lemma = set()
for word in words:
  lemma.add(lemmatizer.lemmatize(word))
print("length after lemmatization is ", len(lemma))

length after lemmatization is  27188


As we can see the count of types in corpus is 30491 and after lemmatization it reduced to 27188 and after stemming it reduced further to 21336

In [99]:
assertTrue(len(lemma) < len(stemmedWords) < len(t.word_counts))

NameError: ignored

In [ ]:
import spacy
nlp = spacy.load("es_core_news_sm")

text = '''Your text here'''
tokens = nlp(text)

for sent in tokens.sents:
    print(sent.string.strip())

